src/app/pages/communication/communication.page.ts

```diff
@@ -1,30 +1,354 @@
+  testCategory: CategoryCode;
 
   constructor(
-    private navController: NavController,
+    public platform: Platform,
+    public router: Router,
+    public authenticationProvider: AuthenticationProvider,
+    store$: Store<StoreModel>,
     public routeByCat: RouteByCategoryProvider,
-  ) { }
+    public deviceAuthenticationProvider: DeviceAuthenticationProvider,
+    private translate: TranslateService,
+  ) {
+    super(platform, router, authenticationProvider, store$);
+    this.form = new FormGroup(this.getFormValidation());
+  }
+
+  ngOnInit(): void {
+    super.ngOnInit();
+    const currentTest$ = this.store$.pipe(
+      select(getTests),
+      select(getCurrentTest),
+    );
+
+    this.pageState = {
+      candidateName$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getCandidateName),
+      ),
+      candidateUntitledName$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getUntitledCandidateName),
+      ),
+      candidateDriverNumber$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getCandidateDriverNumber),
+        map(formatDriverNumber),
+      ),
+      candidateProvidedEmail$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getCandidateEmailAddress),
+        take(1),
+      ),
+      communicationEmail$: this.store$.pipe(
+        select(getTests),
+        select(getCurrentTest),
+        select(getCommunicationPreference),
+        select(getCommunicationPreferenceUpdatedEmail),
+      ),
+      communicationType$: currentTest$.pipe(
+        select(getCommunicationPreference),
+        select(getCommunicationPreferenceType),
+      ),
+      candidateAddress$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getPostalAddress),
+      ),
+      conductedLanguage$: currentTest$.pipe(
+        select(getCommunicationPreference),
+        select(getConductedLanguage),
+      ),
+      testCategory$: currentTest$.pipe(
+        select(getTestCategory),
+        map((result) => this.testCategory = result),
+      ),
+    };
+
+    const {
+      candidateProvidedEmail$,
+      communicationEmail$,
+      communicationType$,
+      conductedLanguage$,
+      testCategory$,
+    } = this.pageState;
+
+    this.merged$ = merge(
+      candidateProvidedEmail$.pipe(map((value) => this.candidateProvidedEmail = value)),
+      communicationEmail$.pipe(map((value) => this.communicationEmail = value)),
+      communicationType$.pipe(map((value) => this.communicationType = value as CommunicationMethod)),
+      conductedLanguage$.pipe(tap((value) => configureI18N(value as Language, this.translate))),
+      testCategory$.pipe(map((result) => this.testCategory = result)),
+    );
+
+    this.subscription = this.merged$.subscribe();
+
+    this.initialiseDefaultSelections();
```

Comment:
in DES3 this is encapsulated within `if (this.shouldPreselectADefaultValue()) {`

Reply:
yes, was redundant